include/swift/Strings.h

```diff
@@ -40,46 +40,76 @@ constexpr static const char MANGLING_MODULE_CLANG_IMPORTER[] =
 /// The name of the Builtin type for Int256
 constexpr static const char BUILTIN_TYPE_NAME_INT256[] = "Builtin.Int256";
+constexpr static const char BUILTIN_TYPE_NAME_INT256_NO_NAMESPACE[] = "Int256";
 /// The name of the Builtin type for Int512
 constexpr static const char BUILTIN_TYPE_NAME_INT512[] = "Builtin.Int512";
+constexpr static const char BUILTIN_TYPE_NAME_INT512_NO_NAMESPACE[] = "Int512";
 /// The name of the Builtin type for IntLiteral
 constexpr static const char BUILTIN_TYPE_NAME_INTLITERAL[] =
     "Builtin.IntLiteral";
-/// The name of the Builtin type for Float
+constexpr static const char BUILTIN_TYPE_NAME_INTLITERAL_NO_NAMESPACE[] =
+    "IntLiteral";
+/// The name of the Builtin type for IEEE Floating point types.
 constexpr static const char BUILTIN_TYPE_NAME_FLOAT[] = "Builtin.FPIEEE";
+constexpr static const char BUILTIN_TYPE_NAME_FLOAT_NO_NAMESPACE[] = "FPIEEE";
+// The name of the builtin type for power pc specific floating point types.
+constexpr static const char BUILTIN_TYPE_NAME_FLOAT_PPC[] = "Builtin.FPPPC";
+constexpr static const char BUILTIN_TYPE_NAME_FLOAT_PPC_NO_NAMESPACE[] =
+    "FPPPC";
 /// The name of the Builtin type for NativeObject
 constexpr static const char BUILTIN_TYPE_NAME_NATIVEOBJECT[] =
     "Builtin.NativeObject";
+constexpr static const char BUILTIN_TYPE_NAME_NATIVEOBJECT_NO_NAMESPACE[] =
+    "NativeObject";
 /// The name of the Builtin type for BridgeObject
 constexpr static const char BUILTIN_TYPE_NAME_BRIDGEOBJECT[] =
     "Builtin.BridgeObject";
+constexpr static const char BUILTIN_TYPE_NAME_BRIDGEOBJECT_NO_NAMESPACE[] =
+    "BridgeObject";
 /// The name of the Builtin type for RawPointer
 constexpr static const char BUILTIN_TYPE_NAME_RAWPOINTER[] =
     "Builtin.RawPointer";
+constexpr static const char BUILTIN_TYPE_NAME_RAWPOINTER_NO_NAMESPACE[] =
+    "RawPointer";
 /// The name of the Builtin type for UnsafeValueBuffer
 constexpr static const char BUILTIN_TYPE_NAME_UNSAFEVALUEBUFFER[] =
     "Builtin.UnsafeValueBuffer";
+constexpr static const char BUILTIN_TYPE_NAME_UNSAFEVALUEBUFFER_NO_NAMESPACE[] =
+    "UnsafeValueBuffer";
 /// The name of the Builtin type for UnknownObject
 constexpr static const char BUILTIN_TYPE_NAME_UNKNOWNOBJECT[] =
     "Builtin.UnknownObject";
+constexpr static const char BUILTIN_TYPE_NAME_UNKNOWNOBJECT_NO_NAMESPACE[] =
+    "UnknownObject";
 /// The name of the Builtin type for Vector
 constexpr static const char BUILTIN_TYPE_NAME_VEC[] = "Builtin.Vec";
+constexpr static const char BUILTIN_TYPE_NAME_VEC_NO_NAMESPACE[] = "Vec";
 /// The name of the Builtin type for SILToken
 constexpr static const char BUILTIN_TYPE_NAME_SILTOKEN[] = "Builtin.SILToken";
+constexpr static const char BUILTIN_TYPE_NAME_SILTOKEN_NO_NAMESPACE[] =
+    "SILToken";
 /// The name of the Builtin type for Word
 constexpr static const char BUILTIN_TYPE_NAME_WORD[] = "Builtin.Word";
+constexpr static const char BUILTIN_TYPE_NAME_WORD_NO_NAMESPACE[] = "Word";
```

Comment:
Can you clean up the duplication here? Either by removing the original constants, or by not adding the new ones and instead adding a fixed offset (`llvm::array_lengthof(BUILTIN_TYPE_NAME_PREFIX)`).

Reply:
I think that would require us to change these to StringLiteral or the like. Last time I did that I ran into problems. Maybe now that works now. Let me try it.

Reply:
Hm, yeah, I see. StringLiteral doesn't have a constructor that would accept an offset. But maybe not everywhere needs to be using StringLiteral either—it's only *really* important for static const / constexpr data.

Reply:
I figured out how to do this using a composition class with StringLiteral. It was pretty difficult to find the exact incantation. Turns out the trick to make it work is to use `{"$STRING"}` constructor. Then in my composition class I have a method that returns a StringRef that is the StringLiteral.drop_front(NUM_CHARS).